photutils/psf/psfphotometry.py

```diff
@@ -0,0 +1,333 @@
+            `~astropy.modeling.fitting` for more details on fitters.
+        niters : int (default=3)
+            Number of iterations to perform the loop FIND, GROUP, SUBTRACT,
+            NSTAR.
+
+        Notes
+        -----
+        If there are problems with fitting large groups, change the parameters
+        of the grouping algorithm to reduce the number of sources in each
+        group or input a ``star_groups`` table that only includes the groups
+        that are relevant (e.g. manually remove all entries that coincide with
+        artifacts).
+        """
+
+        self.find = find
+        self.group = group
+        self.bkg = bkg
+        self.psf = psf
+        self.fitter = fitter
+        self.niters = niters
+        self.fitshape = fitshape
+
+        @property
+        def niters(self):
+            return self._niters
+
+        @niters.setter
+        def niters(self, niters):
+            if isinstance(niters, int) and niters > 0:
+                self._niters = niters
+            else:
+                raise ValueError('niters must be an interger-valued number, '
+                                 'received niters = {}'.format(niters))
+        
+        @property
+        def fitshape(self):
+            return self._fitshape
+
+        @fitshape.setter
+        def fitshape(self, fitshape):
+            fitshape = np.asarray(fitshape)
+            if len(fitshape) == 2:
+                if np.all(fitshape) > 0:
+                    if np.all(fitshape) % 2 == 1:
+                        self._fitshape = fitshape
+                    else:
+                        raise ValueError('fitshape must be odd '
+                                         'integer-valued, '
+                                         'received fitshape = {}'\
+                                         .format(fitshape))
+                else:
+                    raise ValueError('fitshape must have positive elements, '
+                                     'received fitshape = {}'\
+                                     .format(fitshape))
+            else:
+                raise ValueError('fitshape must have two dimensions, '
+                                 'received fitshape = {}'.format(fitshape))
+
+    def __call__(self, image):
+        """
+        Parameters
+        ----------
+        image : 2D array-like, `~astropy.io.fits.ImageHDU`,
+        `~astropy.io.fits.HDUList`
+            Image to perform photometry
+        
+        Returns
+        -------
+        outtab : `~astropy.table.Table`
+            Table with the photometry results, i.e., centroids and fluxes
+            estimations.
+        residual_image : array-like, `~astropy.io.fits.ImageHDU`,
+        `~astropy.io.fits.HDUList`
+            Residual image calculated by subtracting the fitted sources
+            and the original image.
+        """
+
+        return self.do_photometry(image)
+
+    def do_photometry(self, image):
```

Comment:
For list-driven photmetry I suggest to have an optional kwarg where the user specifies the positions. If it's given we don't do source detections. @eteq @hamogu: what do you say?